lib/mocha.js

```diff
@@ -141,7 +141,11 @@ Mocha.prototype.reporter = function(reporter, reporterOptions){
     reporter = reporter || 'spec';
     var _reporter;
     try { _reporter = require('./reporters/' + reporter); } catch (err) {};
-    if (!_reporter) try { _reporter = require(reporter); } catch (err) {};
+    if (!_reporter) try { _reporter = require(reporter); } catch (err) {
+      err.message.indexOf('Cannot find module') !== -1 ?
+        console.warn('"' + reporter + '" reporter not found') :
```

Comment:
IMO, it's better to stay consistent with the same code-style we have today.
e.g: [1](https://github.com/mochajs/mocha/blob/master/lib/reporters/base.js#L111-L114), [2](https://github.com/mochajs/mocha/blob/master/lib/utils.js#L441-L442)